cartofacade/internal/capi/capi.go

```diff
@@ -130,7 +145,7 @@ func New(cfg CartoConfig, acfg CartoAlgoConfig, vcl CartoLib) (Carto, error) {
 
 	vcac := toAlgoConfig(acfg)
 
-	status := C.viam_carto_init(&pVc, vcl.value, vcc, vcac)
+	status := C.viam_carto_init(&pVc, vcl.(*CartoLib).value, vcc, vcac)
```

Comment:
Does this panic if provided a non `CartoLib` concrete type? If so, can we please type assert that vcl has a `CartoLib` concrete type & if not return an error?